src/Config/ObjectModel/RuntimeEntities.cs

```diff
@@ -121,9 +121,28 @@ private static Entity ProcessRestDefaults(Entity nameCorrectedEntity)
         // If no Rest node was provided in the config, set it with the default state of enabled for all verbs
         if (nameCorrectedEntity.Rest is null)
         {
-            nameCorrectedEntity = nameCorrectedEntity
-                with
-            { Rest = new EntityRestOptions(EntityRestOptions.DEFAULT_SUPPORTED_VERBS) };
+            nameCorrectedEntity = (nameCorrectedEntity.Source.Type is EntitySourceType.StoredProcedure) ?
```

Comment:
we can simplify this logic for better readability.
```
    if (nameCorrectedEntity.Rest is null)
    {
        var defaultMethods = (nameCorrectedEntity.Source.Type is EntitySourceType.StoredProcedure) ?
            EntityRestOptions.DEFAULT_SUPPORTED_VERBS_FOR_SP :
            EntityRestOptions.DEFAULT_SUPPORTED_VERBS;

        nameCorrectedEntity = nameCorrectedEntity with { Rest = new EntityRestOptions(defaultMethods) };
    }
    else if (nameCorrectedEntity.Rest.Methods.Length == 0)
    {
        var defaultMethods = (nameCorrectedEntity.Source.Type is EntitySourceType.StoredProcedure) ?
            EntityRestOptions.DEFAULT_SUPPORTED_VERBS_FOR_SP :
            EntityRestOptions.DEFAULT_SUPPORTED_VERBS;

        nameCorrectedEntity = nameCorrectedEntity with
        {
            Rest = new EntityRestOptions(
                Methods: defaultMethods,
                Path: nameCorrectedEntity.Rest.Path,
                Enabled: nameCorrectedEntity.Rest.Enabled
            )
        };
    }
    ```

Reply:
When `nameCorrectedEntity.Rest.Methods.Length == 0` is true, we want to populate the rest methods only for stored procedures (for tables/views, when rest is enabled, we support all 5 request types).

But, I see your point regarding readability, updated the logic.